### PR TITLE
{bio}[foss/2023a] Cellformer v20240917 w/ R 4.3.2

### PR DESCRIPTION
**Pull request by PetrKralCZ**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/395

### CODE REVIEW

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ generoso

**Comment by PetrKralCZ** [comment]
@boegelbot please test @ jsc-zen3

**Review by boegel — APPROVED** [review]
lgtm

File: easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+exts_default_options = {
+    'source_urls': [PYPI_SOURCE],
+    'download_dep_fail': True,
+    'use_pip': True,
+    'sanity_pip_check': True,
+    'installopts': '',
+}
+
+exts_list = [
+    ('typing_extensions', '4.12.2', {
```
Comment (boegel) [inline easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb:53]: There should be no need for this, since `typing-extensions` is already a dependency?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb:53]: @PetrKralCZ Just for sanity's sake: in your test environment, the `scanpy/1.9.8-foss-2023a` module is still loading `hatchling/1.18.0-GCCcore-12.3.0` as a runtime dependency, which includes `typing-extensions` 4.6.3 as an extension. That's wrong, `hatchling` should only ever be a build dependency...

The same goes for the `scanpy` module in my test environment, but there's another difference that affects the order of `$PYTHONPATH`: in my test environment the `pydantic/1.10.13-GCCcore-12.3.0` module is old, it was not rebuilt after the changes in https://github.com/easybuilders/easybuild-easyconfigs/pull/18948/commits/479ecec53a1ac285392b1072181d7ea256d76fa8.

If `pydantic/1.10.13-GCCcore-12.3.0` loads `typing-extensions/4.9.0-GCCcore-12.3.0` (like it is in your case), then it gets superseded by the older `typing-extensions` that is included in the `hatchling` installation that gets loaded as a dependency of `scanpy`.

Long story short: we should make `hatchling` a *build* dependency of `scanpy`, and then the problem should be resolved... => being fixed via #21657

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3124.skitty.os - Linux RHEL 9.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.9.18
See https://gist.github.com/boegel/a7d0a9f5cce726297ead73100889299c for a full test report.

File: easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb (hunk 1 above)
Comment (PetrKralCZ) [reply easybuild/easyconfigs/c/Cellformer/Cellformer-20240917-foss-2023a-R-4.3.2.eb:53]: Other then `scanpy` I also had to rebuild `anndata` on Donphan to pass the pip checks without `typing-extensions` in `exts_list`.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @PetrKralCZ!
